### ISSUE TITLE
[VLSI 82c480] IBM PS/1 model 2133 and [i430LX] Intel Premiere/PC do not start

### ISSUE THREAD

**Issue by maximus105** [86Box/86Box#880]
**Describe the bug**
[VLSI 82c480] IBM PS/1 model 2133 and [i430LX] Intel Premiere/PC do not start at all (freezes) with any type of CPU.

**Desktop (please complete the following information):**
 - Windows 10 x64 2004
 - 86Box v2.10 Build 2408
 - Build type: [optimized]
 - HD controller type: [Internal IDE], C/H/S: 412/16/63 (202 MB)

**Comment by OBattler** [comment]
The PS/1 Model 2133 is now fixed (it was missing Port 92h). Let me look at the Intel Premiere/PCI.

**Comment by OBattler** [comment]
The Intel Premiere/PCI works fine on my end, can I see your 86box.cfg please?

**Comment by Fenix770** [comment]
I tested both machines with the last commit (ver 2.10.0.2414) of standard and dev branch.

The PS/1 2133 still does not boot. Could be I have wrong ROM BIOS files?
The Premier PCI starts correctly and the BIOS detects the correct CPU speed. (P66)

**Comment by WinFutureX** [comment]
PS/1 2133 also doesn't boot for me, POST card shows "00 AA". It did at least write to NVRAM though.

**Comment by ppgrainbow** [comment]
I'm agree with WinFutureX. Also, the SCAMP-based Commodore SL386SX (which uses a 25 MHz 80386SX processor) doesn't boot either. The POST card shows 25 23 despite writing 128 bytes to the NVRAM.

Would it be possible to look into these issues if you can?

Thanks. :)

**Comment by OBattler** [comment]
The SCAMP is broken, I need to fix the chipset emulation.

The PS/1 2133 should work, yes, you could have the wrong BIOS, Fenix770. Look at our roms/ repository.

**Comment by OBattler** [comment]
OK, try the PS/1 2133 now, see if my rewrite of the VLSI chipset fixed anything.

**Comment by maximus105** [comment]
I checked build 2416. The IBM PS/1 model 2133 runs on all CPUs except the iDX4 OverDrive 75 and iDX4 OverDrive 100. (On these CPUs, the system freezes during the POST BIOS phase).
In addition, when the BIOS starts up, a window always appears:

![20200630-155613-130](https://user-images.githubusercontent.com/58270614/86149718-087f1c00-bb05-11ea-9708-d8b17c504aff.png)

In addition, on CPUs:
i486SX / 16
i486SX / 20
i486SX / 25
i486SX / 33
i486DX / 25
i486SX2 / 66 (Q0569)
i486DX / 25
i486DX / 33
i486DX / 50
such games as: Doom, Doom2, Heretic, Hexen do not start (freezes), both when the "Dynamic Recompiler" option is turned on or off. These games run and work without errors only on CPUs:
i486DX2 / 40
i486DX2 / 50
i486DX2 / 66

**Comment by WinFutureX** [comment]
You need to disable cache in BIOS and plug in a PS/2 mouse. Otherwise, it works fine now.

**Comment by maximus105** [comment]
[i430LX] Intel Premiere/PC do not start from under 86Box Manager v1.6.4. Also freezes out and do not start from under 86Box Manager v1.6.4 [OPTi 597] AMI Excalibur VLB. If you do not use 86Box Manager v1.6.4, then these machines start up normally. I don’t know what exactly the problem is: 86Box Manager v1.6.4 or specifically these machines, but I always use 86Box Manager, so I did not immediately understand why these machines freezes.

**Comment by dhrdlicka** [comment]
Does it freeze even if you launch it with the same config and NVRs?

**Comment by nerd73** [comment]
86Box Manager v1.6.4 + AMI Excalibur VLB appears to work fine for me.

**Comment by Fenix770** [comment]
I recreated a Hard disk image from backup disk, avaiable in the site [https://ps1stuff.wordpress.com/](url) (I use the 2133 disks from the fall of 1993)

The recreation finished correctly, but I found a bug with CL-542x drivers (Could be generic?). 

Description of the bugs: 
1.The 256 color modes that use 1mb of vram works fine except, some black lines that apears on background
2. With the 16 color video modes and 256 color video modes that uses 512kb of RAM, the Windows mouse driver does not load.
3. If I launch a DOS window, the DOS prompt does not apears and it looks locked however I can switch the aplication with alt+tab, but I'm unable to close the DOS window. 

4. The cache error appears whether the L1 cache is enabled or disabled. The L2 cache is always 0kb.

Here is the link to the image file: 
[https://drive.google.com/file/d/1Na3XplJi-sGAnadIPzpEPQsQ9YGHwKs7/view?usp=sharing](url)

**Comment by maximus105** [comment]
@dhrdlicka 
Yes. And even if you delete the previously created NVR and 86box.cfg and run 86Box Manager v1.6.4 again, these machines will freeze.

**Comment by OBattler** [comment]
> 1.The 256 color modes that use 1mb of vram works fine except, some black lines that apears on background
That sounds like the 5422/24 drivers, which yes, do that, no idea why.

> 2. With the 16 color video modes and 256 color video modes that uses 512kb of RAM, the Windows mouse driver does not load.
The mouse driver loads, it's just that hardware cursor does not draw. The PCem code has a hint about the cursor stuff, I need figure out what is actually going on.

> 3. If I launch a DOS window, the DOS prompt does not apears and it looks locked however I can switch the aplication with alt+tab, but I'm unable to close the DOS window.
This sounds like bad banking, needs to be looked into.

> 4. The cache error appears whether the L1 cache is enabled or disabled. The L2 cache is always 0kb.
Yes, it's likely caused by the L2 cache, likely something somewhere reports L2 cache as being present even when it's not. It's also present on the Lucky Star 486 machine (the one with the AMI WinBios and SiS 85c496/497), where I had to hack around the issue with a hack in the NVR code that forces external (L2) cache to disabled.

**Comment by OBattler** [comment]
Also, some of these Cirrus bugs I suspect are caused by undocumented / poorly documented features, at least in the 1994 and 1995 techical reference manuals, and the 1993 one is nowhere to be found unfortunately.

**Comment by Fenix770** [comment]
Hello!

I tested again, with the last commit. Some bugs were fixed.

1. The 512k video modes now works right with the mouse
2. The Dos Window now works. The Dos prompt works and exit command closes the window as expected.

Remaining bugs:
Some black lines appears  on the bottom of the screen after closing the Dos window. The Works tutorial closes with a general protection error on 486 DX/33. On 486 SX it simply crashes and closes 86box.

![20200703-143359-262](https://user-images.githubusercontent.com/31600563/86478989-a94a2300-bd21-11ea-94ef-a76f04df63e7.png)
